### PR TITLE
Filter measurements for options applicable to given TSNs

### PR DESCRIPTION
**Pull request by mauberti-bc**
## Links to Jira Tickets

- Supports [SIMSBIOHUB-557](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-557)

## Description of Changes

- Allows measurements to be filtered by TSNs (returns measurements that any of the given TSNs can have)
- Supports https://github.com/bcgov/biohubbc/pull/1338, together with https://github.com/bcgov/biohubbc-platform/pull/250

### CODE REVIEW

File: src/repositories/xref-repository.ts [hunk 1]
```diff
@@ -295,32 +300,44 @@ export class XrefRepository extends Repository {
    *
    * @async
    * @param {IMeasurementSearch} search - Search properties.
+   * @param {number[]} tsns - filters for measurements applied to any of the specified TSNs
    * @returns {Promise<ITsnQualitativeMeasurement[]>}
    */
-  async searchForQualitativeMeasurements(search: IMeasurementSearch): Promise<ITsnQualitativeMeasurement[]> {
+  async searchForQualitativeMeasurements(
+    search: IMeasurementSearch,
```
Comment (MacQSL) [inline src/repositories/xref-repository.ts:307]: We should move the tsns into the search object

File: src/services/xref-service.ts [hunk 2]
```diff
@@ -322,36 +322,28 @@ export class XrefService implements Service {
...
 
-    // Inject the tsn hiearchy into the measurements.
-    const qualitativeWithHiearchy = qualitative.map((measurement) => ({
-      ...measurement,
-      tsnHierarchy: tsnHiearchyMap.get(measurement.itis_tsn) ?? []
-    }));
+    // Fetch TSNs hierarchy if search.tsns exists and has items
+    if (search.tsns?.length) {
+      const tsnsWithHierarchy = await this.itisService.getTsnsHierarchy(search.tsns);
+      tsns = Array.from(new Set(tsnsWithHierarchy.flatMap((tsn) => tsn.hierarchy)));
```
Comment (MacQSL) [inline src/services/xref-service.ts:334]: Does the hierarchy include the intial search tsns?

File: src/repositories/xref-repository.ts (hunk 1 above)
Comment (mauberti-bc) [reply src/repositories/xref-repository.ts]: Done

**Review by MacQSL — APPROVED** [review]
SHIPIT 🚢

**Pull request merged**